src/enclave/enclave_mem.c

```diff
@@ -164,6 +167,22 @@ void* syscall_SYS_mremap(
 
 int syscall_SYS_munmap(void* addr, size_t length)
 {
+    // During thread teardown, libc unmaps the stack of the current thread
+    // before doing an exit system call.  This works on a conventional system
+    // because it's possible to do an exit system call without a stack.  With
+    // LKL, the kernel and userspace share a stack and so any system call needs
+    // a stack.  We work around this by deferring any attempt to unmap the
+    // current stack.
+    register void *rsp __asm__ ("rsp");
+    oe_host_printf("Unmapping: %p + %ld\n", addr, length);
```

Comment:
this is a compilation error for me as oe_host_printf is an implicit declaration.

Reply:
Ooops, that was a debugging line I accidentally left in.